packages/org.openapis.v3/Header.pkl

```diff
@@ -0,0 +1,39 @@
+//===----------------------------------------------------------------------===//
+// Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+/// The Header Object follows the structure of the Parameter Object with the
+/// following changes:
+///
+/// * name MUST NOT be specified, it is given in the corresponding headers map.
+/// * in MUST NOT be specified, it is implicitly in header.
+/// * All traits that are affected by the location MUST be applicable to a 
+///   location of header (for example, style).
+module org.openapis.v3.Header
+
+extends "BaseParameter.pkl"
+
+import "Schema.pkl"
+import "Reference.pkl"
+
+/// Determines whether this parameter is mandatory. 
+required: Boolean?
+
+/// Describes how the parameter value will be serialized depending on the type of the parameter value.
+///
+/// In this case, it's a header, so it should always be "simple".
+style: "simple"?
+
+/// The schema defining the type used for the parameter.
+schema: *Schema|Reference?
```

Comment:
A nullable union needs parentheses around the union type. Otherwise, the `?` is attached only to the last alternative.

```suggestion
schema: (*Schema|Reference)?
```